plugins/de.cognicrypt.staticanalyzer/src/de/cognicrypt/staticanalyzer/handlers/RunAnalysisHandler.java

```diff
@@ -14,13 +24,52 @@
 	public Object execute(ExecutionEvent event) throws ExecutionException {
 		final AnalysisKickOff akf = new AnalysisKickOff();
 		IProject ip = Utils.getCurrentlySelectedIProject();
+		
+//		check if the open editor is in the same project
+		if (ip == Utils.getCurrentProject()) {
+			try {
+//				check if there are unsaved changes
+				if (Utils.getCurrentlyOpenEditor().isDirty()) {
+					int answr = saveFile(Utils.getCurrentlyOpenFile());
+//					save file and analyse
+					if (answr == JOptionPane.NO_OPTION) {
+					}
+//					do not save file and analyse
+					else if(answr == JOptionPane.YES_OPTION) {
+						Utils.getCurrentlyOpenEditor().doSave(null);	
+					}
+//					no analyse no save file
+					else if (answr == JOptionPane.CLOSED_OPTION) {
+						return null;
+					}
+				}
+			}
+			catch (Exception e) {
+				// TODO Auto-generated catch block
+				e.printStackTrace();
+			}
+		}
 		final IJavaElement iJavaElement = JavaCore.create(ip);
 		if (akf.setUp(iJavaElement)) {
 			akf.run();
 		}
 		return null;
 	}
 
+	public static int saveFile(IFile openFileInEditor) throws Exception {

```

Comment:
This should not be part of this PR, should it? Please use branches when working on separate issues.